harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -122,23 +122,67 @@ def _init_device(self) -> None:
             self.device = torch.device("cpu")
         check.is_not_none(self.device)
 
-    def _init_model_and_optimizer(self) -> None:
+    def _init_model_optimizers_and_lr_schedulers(self) -> None:
+        # TODO(DET-3267): add a check for deprecating the old interface when releasing the
+        #       support for multiple optimizers and LR schedulers.
+
         self.context.model = self.trial.build_model()
+        self.context.model = self.context.model.to(self.device)
 
         # TODO: Check that optimizer is not an amp optimizer.
-        self.context.optimizer = self.trial.optimizer(self.context.model)
+        optimizers = self.trial._build_optimizers(self.context.model)
+        if isinstance(optimizers, torch.optim.Optimizer):  # type: ignore
+            self.context.optimizers = [optimizers]
+        elif isinstance(optimizers, list):
+            check.gt_eq(len(optimizers), 1)
+            self.context.optimizers = optimizers
+        else:
+            # TODO(DET-3267): make this a user-facing error when releasing the support
+            #     for multiple optimizers and LR schedulers.
+            raise det.errors.InternalException(
+                "invalid return type from build_optimizers. "
+                "Please return a list or the only instance of "
+                "torch.optim.Optimizer."
+            )
 
-        self._init_device()
-        self.context.model = self.context.model.to(self.device)
+        lr_schedulers = self.trial._build_lr_schedulers(optimizers)
+        if isinstance(lr_schedulers, LRScheduler):
+            self.context.lr_schedulers = [lr_schedulers]
+        elif isinstance(lr_schedulers, list):
+            self.context.lr_schedulers = lr_schedulers
+        else:
+            raise det.errors.InvalidExperimentException(
+                "invalid return type from build_lr_schedulers. "
+                "Please see a list or the only instance of "
+                "determined.pytorch.LRScheduler."
+            )
+        # TODO: clean up _LRHelper
+        self.lr_helpers = [_LRHelper(lr_scheduler) for lr_scheduler in self.context.lr_schedulers]
+
+        def filter_named_parameters(
+            model: nn.Module, optimizer: torch.optim.Optimizer,  # type: ignore
+        ) -> List:
```

Comment:
question: do you think it would be worthwile to write a unit test for this function?

Reply:
IMO, this function is only used here and probably won't be used anywhere else in the future. It's also very transparent and straightforward in what it does. So don't think it's necessary.

Reply:
I have manually tested to make sure it works.